rcl/test/CMakeLists.txt

```diff
@@ -1,10 +1,13 @@
 find_package(ament_cmake_gtest REQUIRED)
 find_package(launch_testing_ament_cmake REQUIRED)
 
-find_package(mimick_vendor REQUIRED)
 find_package(test_msgs REQUIRED)
 
-find_package(mimick_vendor REQUIRED)
+set(USE_MIMICK)
+if(NOT AMENT_IGNORE_MIMICK_TESTS)
+  find_package(mimick_vendor REQUIRED)
```

Comment:
Wait... do you mean creating a target that only has a public compile definition if mimick is being skipped?
See https://github.com/ros2/mimick_vendor/pull/21, for the moment I have added a function.

Reply:
I can create an interface target for that https://cmake.org/cmake/help/latest/command/add_library.html#interface-libraries.
In that case, we cannot rename the compile definition, but that doesn't seem important.
SKIP_MIMICK or MIMICK_VENDOR_SKIP_MIMICK are okay.

Let me know what option you prefer.

Reply:
I have used an interface instead of a function exported by mimick_vendor.

If you want to see the previous approach:

* https://github.com/ros2/rcl/blob/e29cc95387d3975269c575f4fd88bed83e64be01/rcl/test/CMakeLists.txt#L90
* https://github.com/ros2/mimick_vendor/blob/231b35bd1a3824e10c878c66edf9bce68c0a253c/mimick_vendor-extras.cmake.in